src/event-bus.ts

```diff
@@ -65,7 +68,14 @@ export class EventBus<EventBase extends IEvent = IEvent>
 
   bind(handler: IEventHandler<EventBase>, name: string) {
     const stream$ = name ? this.ofEventName(name) : this.subject$;
-    const subscription = stream$.subscribe((event) => handler.handle(event));
+    const subscription = stream$
+      .pipe(mergeMap((event) => from(handler.handle(event))))
+      .subscribe({
+        error: (error) => {
+          this.logger.error(`${handler.constructor.name} produced an error.`);
```

Comment:
@Sikora00 shouldn't it also apply for other *buses?

Reply:
![image](https://user-images.githubusercontent.com/24508253/133035827-1c726789-f8b3-4584-a5c8-407c09168e6f.png)

Reply:
We could do this, but it is not that important there as 

- It doesn't crash the app, because is used mostly in a scope of a request so will be handled by an exception filter and will be mapped to 500 or something.
- It is executed within the main flow, so the user can make a reaction for an exception, which is not possible with events. 

The only place where that comes to my mind is s saga which can produce unhandable execution of the CommandBus 🤔. So because of that, it makes sense, but for QueryBus I don't see such case.

Reply:
And maybe it is not a good idea to make a lot of logs from the framework level. It is reasonable here as there is not other way to recognize such an error, but if we add this to CommandBus often it will look like we are logging the most critical kind of logs which is later properly handled :/

Reply:
I agree that we could also add logging if the command was executed via Saga (and QueryBus doesn't make much sense as you stated)

Reply:
Done @kgajowy